component/useProductList.js

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+
+const useProductList = () => {
+  const [products, setProducts] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [skip, setSkip] = useState(0);
+
+  const fetchProducts = async () => {
+    try {
+      const res = await fetch(`https://dummyjson.com/products?skip=${skip}`);
+      const data = await res.json();
+      setProducts((prevProducts) => [...prevProducts, ...data?.products]);
+    } catch (error) {
+      console.error('Error fetching products:', error);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const loadMore = () => {
+    setSkip((prevSkip) => prevSkip + 5); 
+  };
+
```

Comment:
Need to handle cases like when we reach at end, if we already are hitting pagination api then it should call it again until previous one is completed. Also we need to show activity loader at the end of the footer. Pls do attach video in the description